api/oc_resource_factory.c

```diff
@@ -37,22 +42,89 @@ gen_random_uri(char *uri, size_t uri_length)
   uri[uri_length - 1] = '\0';
 }
 
+unsigned
+get_collection_instance_uri_for_index(oc_collection_t *collection,
+                                      unsigned index, char *uri,
+                                      size_t uri_size)
+{
+  strncpy(uri, oc_string(collection->res.uri),
+          oc_string_len(collection->res.uri));
+  uri[oc_string_len(collection->res.uri)] = '/';
```

Comment:
this can be an issue at windows -> dynamic stack array as I know was not supported.